pydantic/functional_validators.py

```diff
@@ -80,6 +79,10 @@ def __get_pydantic_core_schema__(self, source_type: Any, handler: _GetCoreSchema
             func = cast(core_schema.NoInfoValidatorFunction, self.func)
             return core_schema.no_info_after_validator_function(func, schema=schema)
 
+    @classmethod
+    def _from_decorator(cls, decorator: _decorators.Decorator[_decorators.FieldValidatorDecoratorInfo]) -> Self:
+        return cls(func=decorator.func)
+
```

Comment:
@adriangb, what do you think about this `_from_decorator` pattern? Sort of replaces the old pattern of `apply_validators` using` _VALIDATOR_F_MATCH`. I didn't help to write that section of the codebase, so wanted to get feedback from someone who might have some more thoughts / context.

Reply:
(I'll note that this removes the need for `_VALIDATOR_F_MATCH` and `apply_validators`. They are kept for deprecated decorator code but will be removed in v3).

Reply:
I'm not sure I honestly don't recall any of this

Reply:
No worries. I'm satisfied with this pattern, let's move forward here 👍